src/app/components/Onboarding/domain/useOnboarding.tsx

```diff
@@ -2,17 +2,20 @@ import { ApiClient } from '@shared/infrastructure/http/ApiClient';
 import { DomainHookReturnValue } from '@shared/types/DomainHookReturnValue';
 import { useState } from 'react';
 
-import { useDataPresentCheck } from '../infrastructure/useDataPresentCheck';
+import { useFetchTenantStats } from '../infrastructure/useFetchTenantStats';
 
 export function useOnboarding(): DomainHookReturnValue {
-  const { isFetching, error, hasNoUserData } = useDataPresentCheck();
   const [isModalClosed, setIsModalClosed] = useState(false);
-  const noDatasource = ApiClient.datasourcesCount() === 0;
+
+  const pyroscopeDataSourcesCount = ApiClient.getPyroscopeDataSources().length;
+  const { isFetching, error, stats } = useFetchTenantStats({ enabled: pyroscopeDataSourcesCount > 0 });
```

Comment:
Small refactor to fetch directly from here and not having to pass down `enabled` to `useDataPresentCheck`